cocos/audio/assets/clip.ts

```diff
@@ -28,22 +28,13 @@
  * @module component/audio
  */
 
+import { AudioPlayer } from 'pal/audio';
 import {
     ccclass, type, serializable, override,
 } from 'cc.decorator';
 import { Asset } from '../../core/assets/asset';
-import { Enum } from '../../core/value-types';
-import { AudioPlayer, PlayingState } from './player';
-import { AudioPlayerDOM } from './player-dom';
-import { AudioPlayerWeb } from './player-web';
 import { legacyCC } from '../../core/global-exports';
-
-export const AudioType = Enum({
-    WEB_AUDIO: 0,
-    DOM_AUDIO: 1,
-    JSB_AUDIO: 2,
-    UNKNOWN_AUDIO: 3,
-});
+import { AudioType } from '../../../pal/audio/type';
```

Comment:
Uhh, that is to say, `"pal/audio/type" isn't belong to abstraction layer?

Reply:
no, it should not belong to abstraction layer

another reason is that I can't declare Enum type in .d.ts file :(